pilot/pkg/model/context.go

```diff
@@ -1044,4 +1044,5 @@ func (node *Proxy) IsVM() bool {
 type GatewayController interface {
 	ConfigStoreCache
 	Recompute(GatewayContext) error
+	SetStatusWrite(enabled bool)
```

Comment:
It seems SetStatusWrite is a too narrow scoped function to be in the GatewayController interface. Can we add a generic callback functiond like "OnNewLeader"  - we can have a boolean indicating if this controller is the leader so that we can do these types of things?

Reply:
BTW, Just saw this  https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L165 - So we may not need full interface here though

Reply:
Did a small refactor to avoid needing anything here